extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -52,102 +52,83 @@ const ENDPOINT_INVALID_ERROR =
-  `;
-};
-
-/**
- * Generates the template for each option.
- *
- * @param {!Element} option
- * @return {!Element}
- */
-const buildOptionTemplate = (option) => {
-  const html = htmlFor(option);
-  return html`
-    <span class="i-amphtml-story-quiz-option">
-      <span class="i-amphtml-story-quiz-answer-choice"></span>
-    </span>
-  `;
-};
-
-export class AmpStoryQuiz extends AMP.BaseElement {
+export class AmpStoryReaction extends AMP.BaseElement {
   /**
    * @param {!AmpElement} element
+   * @param {ReactionType} type
    */
-  constructor(element) {
+  constructor(element, type) {
     super(element);
 
-    /** @private @const {!./story-analytics.StoryAnalyticsService} */
-    this.analyticsService_ = getAnalyticsService(this.win, element);
+    /** @protected @const {ReactionType} */
+    this.reactionType_ = type;
 
-    /** @private {!Array<string>} */
-    this.answerChoiceOptions_ = ['A', 'B', 'C', 'D'];
+    /** @protected @const {!./story-analytics.StoryAnalyticsService} */
+    this.analyticsService_ = getAnalyticsService(this.win, element);
 
-    /** @private {?Promise<!../../../src/service/cid-impl.CidDef>} */
+    /** @protected {?Promise<!../../../src/service/cid-impl.CidDef>} */
     this.clientIdService_ = Services.cidForDoc(this.element);
 
-    /** @private {?Promise<JsonObject>} */
+    /** @protected {?Promise<JsonObject>} */
     this.clientIdPromise_ = null;
 
-    /** @private {boolean} */
+    /** @protected {boolean} */
     this.hasUserSelection_ = false;
 
-    /** @private {!../../../src/service/localization.LocalizationService} */
-    this.localizationService_ = Services.localizationService(this.win);
+    /** @protected {?Element} */
+    this.rootEl_ = null;
 
-    /** @private {?Element} */
-    this.quizEl_ = null;
-
-    /** @private {?string} */
+    /** @protected {?string} */
     this.reactionId_ = null;
 
-    /** @private {!./amp-story-request-service.AmpStoryRequestService} */
+    /** @protected {!./amp-story-request-service.AmpStoryRequestService} */
     this.requestService_ = getRequestService(this.win, this.element);
 
-    /** @private {?Promise<?ReactionResponseType|?JsonObject|undefined>} */
+    /** @protected {?Promise<?ReactionResponseType|?JsonObject|undefined>} */
     this.responseDataPromise_ = null;
 
-    /** @private {?ReactionResponseType} */
+    /** @protected {?ReactionResponseType} */
     this.responseData_ = null;
 
-    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    /** @const @protected {!./amp-story-store-service.AmpStoryStoreService} */
     this.storeService_ = getStoreService(this.win);
 
-    /** @const @private {!./variable-service.AmpStoryVariableService} */
+    /** @const @protected {!./variable-service.AmpStoryVariableService} */
     this.variableService_ = getVariableService(this.win);
   }
 
   /** @override */
   buildCallback() {
-    this.quizEl_ = buildQuizTemplate(this.element);
+    this.buildComponent(this.element);
```

Comment:
Abstracted building component into a method that gets implemented by concrete classes

Reply:
👍